src/platform_impl/ios/uikit/gesture_recognizer.rs

```diff
@@ -0,0 +1,147 @@
+use icrate::Foundation::{CGFloat, NSInteger, NSObject, NSUInteger};
+use objc2::{
+    encode::{Encode, Encoding},
+    extern_class, extern_methods, msg_send_id, mutability,
+    rc::Id,
+    runtime::{AnyObject, Sel},
+    ClassType,
+};
+
+use super::UIView;
+
+// https://developer.apple.com/documentation/uikit/uigesturerecognizer
+extern_class!(
+    #[derive(Debug, PartialEq, Eq, Hash)]
+    pub(crate) struct UIGestureRecognizer;
+
+    unsafe impl ClassType for UIGestureRecognizer {
+        type Super = NSObject;
+        type Mutability = mutability::InteriorMutable;
+    }
+);
+
+extern_methods!(
+    unsafe impl UIGestureRecognizer {
+        #[method(state)]
+        pub fn state(&self) -> UIGestureRecognizerState;
+
+        #[method(initWithTarget:action:)]
+        pub unsafe fn initWithTarget_action(&self, target: &AnyObject, action: Sel);
+    }
+);
+
+unsafe impl Encode for UIGestureRecognizer {
+    const ENCODING: Encoding = Encoding::Object;
+}
+
+// https://developer.apple.com/documentation/uikit/uigesturerecognizer/state
+#[repr(transparent)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct UIGestureRecognizerState(NSInteger);
+
+unsafe impl Encode for UIGestureRecognizerState {
+    const ENCODING: Encoding = NSInteger::ENCODING;
+}
+
+#[allow(dead_code)]
+impl UIGestureRecognizerState {
+    pub const Possible: Self = Self(0);
+    pub const Began: Self = Self(1);
+    pub const Changed: Self = Self(2);
+    pub const Ended: Self = Self(3);
+    pub const Cancelled: Self = Self(4);
+    pub const Failed: Self = Self(5);
+}
+
+// https://developer.apple.com/documentation/uikit/uipinchgesturerecognizer
+extern_class!(
+    #[derive(Debug, PartialEq, Eq, Hash)]
+    pub(crate) struct UIPinchGestureRecognizer;
+
+    unsafe impl ClassType for UIPinchGestureRecognizer {
+        type Super = UIGestureRecognizer;
+        type Mutability = mutability::InteriorMutable;
+    }
+);
+
+extern_methods!(
+    unsafe impl UIPinchGestureRecognizer {
+        #[method(scale)]
+        pub fn scale(&self) -> CGFloat;
+
+        #[method(velocity)]
+        pub fn velocity(&self) -> CGFloat;
+    }
+);
+
+unsafe impl Encode for UIPinchGestureRecognizer {
+    const ENCODING: Encoding = Encoding::Object;
+}
+
+impl UIPinchGestureRecognizer {
+    pub(crate) fn init_with_target(target: &UIView, action: Sel) -> Id<Self> {
+        unsafe { msg_send_id![Self::alloc(), initWithTarget: target, action: action] }
+    }
+}
```

Comment:
I'd prefer it like this, as it'll make it easier for me to replace with `icrate`.

```suggestion
extern_methods!(
    unsafe impl UIPinchGestureRecognizer {
        #[method(scale)]
        pub fn scale(&self) -> CGFloat;

        #[method(velocity)]
        pub fn velocity(&self) -> CGFloat;
        
        #[method_id(initWithTarget:action:)]
        pub unsafe fn initWithTarget_action(&self, target: &AnyObject, action: Sel) -> Id<Self>;
    }
);

unsafe impl Encode for UIPinchGestureRecognizer {
    const ENCODING: Encoding = Encoding::Object;
}
```

Same goes for the others.